### DISCUSSION TITLE
Invalid query - invalid identifier parts when using CASE

### DISCUSSION THREAD

**Discussion by masterJedi21** [pocketbase/pocketbase#2759] [Q&A] [answered]
First of all, thanks for this amazing project Gani, you are a legend!
Rookie here so I'm pretty sure I'm doing something wrong. Here is my query:

`select 
  (ROW_NUMBER() OVER()) as id,
  b.id as bookId,
  b.title,
  b.author,
  b.pages,
  SUM(r.pages_read) as pages_read,
  b.review, 
  b.reader,
  b.collection,
  b.cover,
  case
      when sum(r.pages_read) = b.pages then 'completed'
      when sum(r.pages_read) > 0 and sum(r.pages_read) < b.pages then 'in progress'
      when r.pages_read is null then 'not started'
  end as status

from books b 
left join reading r on b.id = r.book

group by b.id, b.title, b.author, b.pages, b.review, b.reader, b.collection, b.cover`

Even using a simple case like the following throws the same error:

`case
    when b.pages > 100 then "long"
    else "short"
end as test`

Thanks in advance for your help.

**Comment by ganigeorgiev** [comment, marked as answer]
You can try wrapping the `CASE` statement in parenthesis, eg. `(CASE ...) as status`.

**Reply by masterJedi21** [reply]
It worked perfectly! Thanks Gani!
